pkg/minikube/download/download_test.go

```diff
@@ -183,41 +183,60 @@ func testImageToCache(t *testing.T) {
+	gcsCheckCalls := 0
+	ghCheckCalls := 0
+	originalGCSCheck := checkRemotePreloadExistsGCS
+	originalGHCheck := checkRemotePreloadExistsGitHub
+	preloadStates = make(map[string]map[string]preloadState)
+	checkRemotePreloadExistsGCS = func(_, _ string) bool {
+		gcsCheckCalls++
 		return doesPreloadExist
 	}
+	checkRemotePreloadExistsGitHub = func(_, _ string) bool {
+		ghCheckCalls++
+		return false
+	}
+	t.Cleanup(func() {
+		checkRemotePreloadExistsGCS = originalGCSCheck
+		checkRemotePreloadExistsGitHub = originalGHCheck
+		preloadStates = make(map[string]map[string]preloadState)
+	})
+
 	existence := PreloadExists("v1", "c1", "docker", true)
-	if existence || !checkCalled {
-		t.Errorf("Expected preload not to exist and a check to be performed. Existence: %v, Check: %v", existence, checkCalled)
+	if existence || gcsCheckCalls != 1 || ghCheckCalls != 1 {
+		t.Errorf("Expected preload not to exist and checks to be performed. Existence: %v, GCS Calls: %d, GH Calls: %d", existence, gcsCheckCalls, ghCheckCalls)
 	}
-	checkCalled = false
+	gcsCheckCalls = 0
+	ghCheckCalls = 0
 	existence = PreloadExists("v1", "c1", "docker", true)
-	if existence || checkCalled {
-		t.Errorf("Expected preload not to exist and no check to be performed. Existence: %v, Check: %v", existence, checkCalled)
+	if existence || gcsCheckCalls != 0 || ghCheckCalls != 0 {
+		t.Errorf("Expected preload not to exist and no checks to be performed. Existence: %v, GCS Calls: %d, GH Calls: %d", existence, gcsCheckCalls, ghCheckCalls)
 	}
 	doesPreloadExist = true
-	checkCalled = false
+	gcsCheckCalls = 0
+	ghCheckCalls = 0
 	existence = PreloadExists("v2", "c1", "docker", true)
-	if !existence || !checkCalled {
-		t.Errorf("Expected preload to exist and a check to be performed. Existence: %v, Check: %v", existence, checkCalled)
+	if !existence || gcsCheckCalls != 1 || ghCheckCalls != 0 {
+		t.Errorf("Expected preload to exist via GCS. Existence: %v, GCS Calls: %d, GH Calls: %d", existence, gcsCheckCalls, ghCheckCalls)
 	}
-	checkCalled = false
+	gcsCheckCalls = 0
+	ghCheckCalls = 0
 	existence = PreloadExists("v2", "c2", "docker", true)
-	if !existence || !checkCalled {
-		t.Errorf("Expected preload to exist and a check to be performed. Existence: %v, Check: %v", existence, checkCalled)
+	if !existence || gcsCheckCalls != 1 || ghCheckCalls != 0 {
+		t.Errorf("Expected preload to exist via GCS for new runtime. Existence: %v, GCS Calls: %d, GH Calls: %d", existence, gcsCheckCalls, ghCheckCalls)
 	}
-	checkCalled = false
+	gcsCheckCalls = 0
+	ghCheckCalls = 0
 	existence = PreloadExists("v2", "c2", "docker", true)
-	if !existence || checkCalled {
-		t.Errorf("Expected preload to exist and no check to be performed. Existence: %v, Check: %v", existence, checkCalled)
+	if !existence || gcsCheckCalls != 0 || ghCheckCalls != 0 {
+		t.Errorf("Expected preload to exist and no checks to be performed. Existence: %v, GCS Calls: %d, GH Calls: %d", existence, gcsCheckCalls, ghCheckCalls)
```

Comment:
These tests are too intrusive. We need to have more high level tests that test the system behavior instead of the implementation details.

The issues exist before this change but hits change makes it worse.